### PR TITLE
feat: scrollable and centered heatmap

### PR DESCRIPTION
**Pull request by iori-yja**
When the heatmap is wider than window size, the heatmap is horizontally scrollable. If not, it is centered.

### CODE REVIEW

**Comment by spiermar** [comment]
I like the solution of scrolling just the heatmap, so we don't have to reduce the size. @brendangregg, any thoughts? This change basically adds a horizontal scrollbar to the heatmap (but just the heatmap, not the window), so we don't have to reduce the size when the profile is longer (120s+). 

Not a big fan of the `overriding` the default legend with a `width: 0` and rewriting the same function here. This would probably be better handled by a change in the plugin to add this as an option. Created https://github.com/spiermar/d3-heatmap2/issues/31 to track it.

File: src/components/Heatmap/heatmap.less [hunk 1]
```diff
@@ -4,12 +4,30 @@ text {
...
     position: relative;
     left: 50%;
-    -webkit-transform: translateX(-50%);
-    -ms-transform: translateX(-50%);
     transform: translateX(-50%);
+    position: relative;
+    text-align: center;
+}
+
+.legend {
```
[the remark this answers is not in the document]
Comment (iori-yja) [reply src/components/Heatmap/heatmap.less]: It does not construct legend but just _moving out_. So the legend is still needed to be rendered by the heatmap. The reason of `width: 0px` is not to hide it but for position alignment. I think I could do better to align it, though.

File: src/components/Heatmap/heatmap.less [hunk 2]
```diff
@@ -4,12 +4,30 @@ text {
...
+    width: 0px;
+    margin: auto;
+    display: block;
+    padding-top: 60px;
+}
+
+.heatmap svg {
```
[the remark this answers is not in the document]
Comment (iori-yja) [reply src/components/Heatmap/heatmap.less]: No. This is my fault to leave both of them. I will delete this.

File: src/components/Heatmap/heatmap.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (iori-yja) [reply src/components/Heatmap/heatmap.less]: Oh, it turn out that `width: 0px` isn't needed.

File: src/components/Heatmap/heatmap.less [hunk 3]
```diff
@@ -4,12 +4,30 @@ text {
...
 }
 
-.heatmap svg {
+.heatmap {
+    overflow-x: auto;
+    width: ~"calc(100vw - 64px)";
+    margin: 0 auto;
     position: relative;
     left: 50%;
-    -webkit-transform: translateX(-50%);
```
[the remark this answers is not in the document]
Comment (iori-yja) [reply src/components/Heatmap/heatmap.less]: That's correct but I forgot to check it on Edge by my own. I will.

**Comment by iori-yja** [comment]
I love https://github.com/spiermar/d3-heatmap2/issues/31. Having `draw-legend-on` option in d3-heatmap is a good idea and I once thought to make a similar issue. I did this hack just for my own purpose.

**Comment by brendangregg** [comment]
I'm happy with this, thanks. I noticed that the Google Chrome browser already gives me a horizontal scrollbar, however, Firefox does not (which this PR fixes).

**Comment by spiermar** [comment]
> I love [spiermar/d3-heatmap2#31](https://github.com/spiermar/d3-heatmap2/issues/31). Having `draw-legend-on` option in d3-heatmap is a good idea and I once thought to make a similar issue. I did this hack just for my own purpose.

I'm open to pull requests on https://github.com/spiermar/d3-heatmap2/issues/31, or I'll try to implement it this week myself, than you could try to use the new feature instead. Up to you.

File: src/components/Heatmap/heatmap.less (hunk 3 above)
[the remark this answers is not in the document]
Comment (iori-yja) [reply src/components/Heatmap/heatmap.less]: I've checked with Edge. It works without `ms` or `webkit` vendor prefix.

**Comment by spiermar** [comment]
@iori-yja can you try `master` of `d3-heatmap2`? The last 2 commits in the repo should take care of rendering the legend outside the main SVG. Basically is just creating a new `div` and setting a property in the script.

The main example covers the usage.

https://github.com/spiermar/d3-heatmap2/blob/master/examples/index.html#L157

Let me know if you have any questions. If it works, let me know if you have any issues changing the PR to take advantage of that.

**Comment by iori-yja** [comment]
Thank you! I have updated the branch to make it work fine. Could you publish newer version of d3-heatmap2 before merging to keep everyone able to build flamescope.

File: src/components/Heatmap/heatmap.less [hunk 4]
```diff
@@ -4,12 +4,24 @@ text {
...
-    -ms-transform: translateX(-50%);
     transform: translateX(-50%);
+    position: relative;
+    text-align: center;
+}
+
+.legend {
+    width: 300px;
```
Comment (spiermar) [inline src/components/Heatmap/heatmap.less]: Container needs to be a little bit bigger than the legend itself, at least 16px, for the SVG overflow.

**Comment by iori-yja** [comment]
Thank you! I've updated with w = 316px, h = 80px.

**Comment by spiermar** [comment]
@iori-yja merged. Thank you!

**Pull request merged**